In [56]:
import numpy as np
from scipy import stats
import pandas as pd

In [57]:
#Ejecicio 1: encuentra el data set 

unis = pd.read_csv("./Estadística/uk_universities.csv")
unis

# Ejercicio 2:
# Extraer los siguentes estadísiticos del conjunto de datos para varias de las variables disponibles:


University_name              Region  Founded_year  \
0           University of Cambridge     East of England          1209   
1              University of Oxford  South East England          1096   
2          University of St Andrews            Scotland          1413   
3           Imperial College London              London          1907   
4           Loughborough University       East Midlands          1966   
..                              ...                 ...           ...   
126       University of East London              London          1992   
127      University of Bedfordshire     East of England          2006   
128           University of Suffolk     East of England          2007   
129      Wrexham Glyndwr University               Wales          2008   
130  Ravensbourne University London              London          1962   

                                                 Motto  UK_rank  World_rank  \
0                 From here, light and sacred draughts        1           4   
1                                 The Lord is my light        2           2   
2                                        Ever to excel        3          86   
3    Knowledge is the adornment and safeguard of th...        4           8   
4                     With Truth, Knowledge and Labour        5         404   
..                                                 ...      ...         ...   
126                    Science and fulfillment of vows      127         971   
127                                                NaN      128        1281   
128                              Honesty and diligence      129        4030   
129                       Confidence through Education      130        2397   
130                              Designed for industry      131        2759   

     CWUR_score  Minimum_IELTS_score  UG_average_fees_(in_pounds)  \
0          94.1                  6.5                        21750   
1          93.3                  6.5                        21770   
2          75.8                  6.5                        17040   
3          86.6                  6.5                        23500   
4          72.8                  5.5                        16400   
..          ...                  ...                          ...   
126         NaN                  4.5                        10700   
127         NaN                  4.5                         9750   
128         NaN                  4.5                         9495   
129         NaN                  4.5                        10250   
130         NaN                  5.5                         9250   

     PG_average_fees_(in_pounds)  ... Student_satisfaction Student_enrollment  \
0                          23187  ...               85.50%      20,000-24,999   
1                          19888  ...               86.50%      25,000-29,999   
2                          15440  ...               87.90%      10,000-14,999   
3                          29900  ...               77.90%      15,000-19,999   
4                          16400  ...               85.80%      15,000-19,999   
..                           ...  ...                  ...                ...   
126                        11700  ...               76.10%      10,000-14,999   
127                         9900  ...               77.40%      10,000-14,999   
128                         9495  ...               76.00%        7,000-7,999   
129                        10500  ...               74.30%        5,000-5,999   
130                         9250  ...               66.10%        2,000-2,999   

    Academic_staff Control_type Academic_Calender Campus_setting  \
0       over-5,000       Public        Trimesters          Urban   
1       over-5,000       Public        Trimesters          Urban   
2      1,000-1,499       Public         Semesters       Suburban   
3      4,000-4,499       Public        Continuous          Urban   
4      1,500-1,999       Public         Semesters   

In [58]:
unis.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 131 entries, 0 to 130
Data columns (total 21 columns):
 #   Column                                         Non-Null Count  Dtype  
---  ------                                         --------------  -----  
 0   University_name                                131 non-null    object 
 1   Region                                         131 non-null    object 
 2   Founded_year                                   131 non-null    int64  
 3   Motto                                          114 non-null    object 
 4   UK_rank                                        131 non-null    int64  
 5   World_rank                                     131 non-null    int64  
 6   CWUR_score                                     84 non-null     float64
 7   Minimum_IELTS_score                            131 non-null    float64
 8   UG_average_fees_(in_pounds)                    131 non-null    int64  
 9   PG_average_fees_(in_pounds)                    131 non

In [59]:
unis.shape

(131, 21)

In [60]:
precio = unis["UG_average_fees_(in_pounds)"]
precio.unique()


array([21750, 21770, 17040, 23500, 16400, 14900, 14940, 15660, 15820,
       14300, 15500, 13500, 13860, 15200, 14500,  8500, 13200, 14660,
       14140, 17890, 14200, 12950, 15850, 13000, 12680, 13850, 15600,
       13800, 13300, 13020, 14000, 13280, 14250, 12500, 14450, 14350,
       12163, 11870, 16880, 10500, 11275, 13650, 10766, 12800, 10650,
       12890, 11150, 11500, 11250, 12495, 10800, 12000, 12400, 18630,
       12710, 10250, 11750, 11450, 12250, 12300, 11000, 15950, 10000,
       10700, 10850, 10750, 10380, 10900, 17160, 11120, 10467, 11300,
        9250, 12100, 11200,  8350, 11795,  9500, 10100, 12660, 11400,
        9750,  9495])

In [61]:
unis["World_rank"].value_counts()

196     2
4       1
2386    1
2190    1
1284    1
       ..
607     1
164     1
140     1
403     1
2759    1
Name: World_rank, Length: 130, dtype: int64

In [62]:
unis["World_rank"].unique()





array([   4,    2,   86,    8,  404,   23,   70,  130,   14,   80,  261,
         93,  133,  119,   69,   76, 2840,   56,  129,  110,  143,  128,
        149,  185,   24,   97,  172,  196,  157,   27,  422,  342,  314,
        182,  200,  384,  403,  140,  164,  607,  167, 2030, 3420,  672,
        466,  729,  659,  699,   98,  611,  877, 2319, 1184,  416, 1560,
        153,  531,  469,  634, 2103, 1111,  760, 4300,  674,  803,  576,
        654, 1273,  542, 2585, 2943,  301,  456,  804, 4608,  734,  691,
       1579,  459, 2643, 2494, 1163,  747, 2386, 1040, 1341,  604, 4336,
       1682, 1140, 2004, 2346,  782, 2073,  573, 1284, 2190, 1066,  578,
        676, 1855, 4547, 3916, 1392,  817,  537, 1905, 1050, 1334,  680,
       1821, 2727, 1129, 5253, 1198, 3200, 1578,  801,  730, 2333,  761,
        657,  864, 2100, 1025,  971, 1281, 4030, 2397, 2759])

In [63]:
unis.describe()

Founded_year     UK_rank  World_rank  CWUR_score  Minimum_IELTS_score  \
count    131.000000  131.000000   131.00000   84.000000           131.000000   
mean    1925.221374   65.900763  1097.59542   73.960714             4.946565   
std      139.040187   37.922158  1145.00106    6.351473             0.617850   
min     1096.000000    1.000000     2.00000   65.900000             4.000000   
25%     1896.000000   33.500000   196.00000   68.675000             4.500000   
50%     1966.000000   65.000000   691.00000   72.600000             4.500000   
75%     1992.000000   98.000000  1569.00000   77.700000             5.500000   
max     2013.000000  131.000000  5253.00000   94.100000             6.500000   

       UG_average_fees_(in_pounds)  PG_average_fees_(in_pounds)  \
count                   131.000000                   131.000000   
mean                  12537.450382                 13099.763359   
std                    2555.318861                  3106.398149   
min                    8350.000000                  6000.000000   
25%                   10758.000000                 11250.000000   
50%                   12000.000000                 12500.000000   
75%                   13825.000000                 14182.500000   
max                   23500.000000                 29900.000000   

       Estimated_cost_of_living_per_year_(in_pounds)    Latitude   Longitude  
count                                     131.000000  131.000000  131.000000  
mean                                     8644.732824   52.749708   -1.673287  
std                                      1579.049538    1.638257    1.503049  
min                                      5500.000000   50.171300   -6.675200  
25%                                      7710.500000   51.499700   -2.748000  
50%                                      8217.000000   52.407400   -1.551600  
75%                                      9457.500000   53.608500   -0.388150  
max                                     15000.000000   57.164800    1.296700

In [64]:
# Media:


rank_media = np.mean(unis["World_rank"])
precio_media = np.mean(precio)


In [65]:
# Desviación con respecto a la media

datos = unis["World_rank"]
rank_desv_media = np.abs(datos-rank_media)


In [66]:
precio_desv_media = np.abs(precio-precio_media)

In [67]:
# Varianza

rank_var = np.var(datos)
precio_var = np.var(precio)
precio_var

6479809.789522755

In [68]:
# Desviación tipica
rank_desv_tipica = np.std(datos)
precio_desv_tipica = np.std(precio)

In [69]:
# Moda

rank_moda = stats.mode(datos)
precio_moda = stats.mode(precio)


In [70]:
# Mediana
rank_mediana = np.median(datos)
precio_mediana = np.median(precio)

In [71]:
# Media ponderada

# En este caso le damos el mismo peso a todos los valores. No son datos dónde tenga sentido aplicar una media ponderada.
# Necesitamos un peso para cada elemento que componga la media ponderada (5 en este caso)
# Calculamos la media ponderada usando el método average() de Numpy

w = np.ones(131, dtype=np.complex64)


rank_media_ponderada = np.average(datos, weights = w)
precio_media_ponderada = np.average(precio, weights = w)


In [72]:
# Coeficiente de variación

rank_coef_var = np.std(datos) / np.abs(np.mean(datos)) # coeficiente de variación

precio_coef_var = np.std(precio) / np.abs(np.mean(precio))

VARIABLE RANK_WORLD

En Uk tenemos la segunda mejor universidad del mundo, y también algunas que están en el puesto 4000 en adelante. 
De media, las universidades de UK están en el puesto 1097 aunque cuentan con 4 de entre las 15 mejores del mundo. 

A priori creímos que la moda no tendría sentido de estudio, pero hemos comprobado que tenemos dos universidades empatadas en un puesto de lista. Hay dos universidades en el ranking 196.

No le vemos el sentido, a priori, de obtener la media ponderada de estos datos en concreto. 



VARIABLE PRECIO UNIVERSIDAD

Los precios de las universidades de UK varían entre 8350 y 23500 pounds.
La media se establece cerca de 12540, y la mediana en 12000 por lo que deducimos unos datos muy equilibrados ¿¿¿¿????

Cerca de la mitad de la tabla, un poco por denajo se encuentra la moda, con 7 universidades cuyo precio es de 10500 pounds.

Cómo en el otro conjuto, no le vemos sentido a realizar la media ponderada de los datos de precio.


In [81]:
precio.min()

8350

In [82]:
precio.max()

23500

In [73]:
precio_coef_var

0.20303546364186892

In [74]:
precio_desv_media

0       9212.549618
1       9232.549618
2       4502.549618
3      10962.549618
4       3862.549618
           ...     
126     1837.450382
127     2787.450382
128     3042.450382
129     2287.450382
130     3287.450382
Name: UG_average_fees_(in_pounds), Length: 131, dtype: float64

In [75]:
precio_desv_tipica

2545.5470511312014

In [76]:
precio_media

12537.45038167939

In [77]:
precio_media_ponderada

(12537.45038167939+0j)

In [78]:
precio_mediana

12000.0

In [79]:
precio_moda

ModeResult(mode=array([10500]), count=array([7]))

In [80]:
precio_var

6479809.789522755